internal/args/errors.go

```diff
@@ -0,0 +1,167 @@
+package args
+
+import (
+	"fmt"
+	"reflect"
+)
+
+//
+// Marshal & Unmarshal Errors
+//
+
+type DataMustBeAPointerError struct {
+}
+
+func (e *DataMustBeAPointerError) Error() string {
+	return fmt.Sprintf("data must be a pointer to a struct")
+}
+
+type UnknownKindError struct {
+	Kind reflect.Kind
+}
+
+func (e *UnknownKindError) Error() string {
+	return fmt.Sprintf("unknown kind '%s'", e.Kind)
+}
+
+//
+// Marshal Errors
+//
+
+type DataMustBeAmArshalableValueError struct {
+}
+
+func (e *DataMustBeAmArshalableValueError) Error() string {
+	return fmt.Sprintf("data must be a marshable value (a scalar type or a Marshaler)")
```

Comment:
Should not use struct for static message
```
ErrDataMustBeAmArshalableValue = fmt.Errorf("data must be a marshable value (a scalar type or a Marshaler)")
```

Reply:
I thought we agreed to use struct for all errors
> All errors, whether they are useful for the end user or not, should follow the format presented bellow.
> pro: we don't need to think about how to group errors, and can concentrate on more important problems
> pro: we can use a single switch to reformat some errors if needed